### PR TITLE
V2 registry mirroring

### PR DESCRIPTION
**Pull request by dmcgowan**
Use the first specified mirror when pulling in v2

### CODE REVIEW

**Comment by stevvooe** [comment]
I'm going to wait for the other PRs to get merged before reviewing this.

**Comment by stevvooe** [comment]
@dmcgowan What happened to this PR? Seems to be the entire client diff.

**Comment by dmcgowan** [comment]
Rebased

File: registry/session_v2.go [hunk 1]
```diff
@@ -19,10 +20,27 @@ func getV2Builder(e *Endpoint) *v2.URLBuilder {
 	return e.URLBuilder
 }
 
-func (r *Session) V2RegistryEndpoint(index *IndexInfo) (ep *Endpoint, err error) {
-	// TODO check if should use Mirror
-	if index.Official {
-		ep, err = newEndpoint(REGISTRYSERVER, true)
+func (r *Session) V2RegistryEndpoint(index *IndexInfo, useMirror bool) (ep *Endpoint, err error) {
+	if useMirror && len(index.Mirrors) > 1 {
```
Comment (stevvooe) [inline registry/session_v2.go]: Should this be `len(index.Mirrors) > 1`?

**Comment by dmcgowan** [comment]
Closing in favor of https://github.com/docker/docker/pull/10201

**Pull request closed without merging**